server/routes/api/reactions/reactions.ts

```diff
@@ -0,0 +1,45 @@
+import Router from "koa-router";
+import auth from "@server/middlewares/authentication";
+import validate from "@server/middlewares/validate";
+import { Comment, Reaction } from "@server/models";
+import { authorize } from "@server/policies";
+import { presentReaction } from "@server/presenters";
+import { APIContext } from "@server/types";
+import * as T from "./schema";
+
+const router = new Router();
+
+router.post(
+  "reactions.list",
+  auth(),
+  validate(T.ReactionsListSchema),
+  async (ctx: APIContext<T.ReactionsListReq>) => {
+    const { commentId } = ctx.input.body;
+    const { user } = ctx.state.auth;
+
+    const comment = await Comment.findByPk(commentId, {
+      rejectOnEmpty: true,
+    });
+
+    authorize(user, "readReaction", comment);
+
+    const reactions: Reaction[] = [];
+
+    await Reaction.findAllInBatches<Reaction>(
```

Comment:
I think this should use the pagination pattern instead, you can use `fetchAll` on the client to make the usage there the same.

Reply:
I get what you mean here. However we don't have a store for reactions; can't use `fetchAll` :(
Let me implement pagination in the `comment.fetchReactionsData()` on the client..